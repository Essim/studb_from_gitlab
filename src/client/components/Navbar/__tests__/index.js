import React from 'react';
import {Navbar} from '../';
import { shallowToJson } from 'enzyme-to-json';

const mockDivision= {
        loading: false,
        divisions: [
            {
                _id: '5a17f595977b356a3d6d3918',
                acronym: 'BIN',
                cuid: 'cuiddivision1',
                logicalDelete: false,
                __v: 0,
                courses: [
                    {
                        name: 'Anglais',
                        grade: 1,
                        logicalDelete: false,
                        cuid: 'anglais-1',
                        _id: '5a17f595977b356a3d6d3920'
                    },
                    {
                        name: 'Web',
                        grade: 1,
                        logicalDelete: false,
                        cuid: 'web-1',
                        _id: '5a17f595977b356a3d6d391f'
                    },
                    {
                        name: 'Assembleur',
                        grade: 1,
                        logicalDelete: false,
                        cuid: 'assembly-1',
                        _id: '5a17f595977b356a3d6d391e'
                    },
                    {
                        name: 'Anglais',
                        grade: 2,
                        logicalDelete: false,
                        cuid: 'anglais-2',
                        _id: '5a17f595977b356a3d6d391d'
                    },
                    {
                        name: 'Web',
                        grade: 2,
                        logicalDelete: false,
                        cuid: 'web-2',
                        _id: '5a17f595977b356a3d6d391c'
                    },
                    {
                        name: 'Language C',
                        grade: 2,
                        logicalDelete: false,
                        cuid: 'c-lang-1',
                        _id: '5a17f595977b356a3d6d391b'
                    },
                    {
                        name: 'Anglais',
                        grade: 3,
                        logicalDelete: false,
                        cuid: 'anglais-3',
                        _id: '5a17f595977b356a3d6d391a'
                    },
                    {
                        name: 'Web',
                        grade: 3,
                        logicalDelete: false,
                        cuid: 'web-3',
                        _id: '5a17f595977b356a3d6d3919'
                    }
                ],
                grades: [
                    null,
                    {
                        name: '1BIN',
                        grade: 1,
                        courses: [
                            {
                                name: 'Anglais',
                                grade: 1,
                                logicalDelete: false,
                                cuid: 'anglais-1',
                                _id: '5a17f595977b356a3d6d3920'
                            },
                            {
                                name: 'Web',
                                grade: 1,
                                logicalDelete: false,
                                cuid: 'web-1',
                                _id: '5a17f595977b356a3d6d391f'
                            },
                            {
                                name: 'Assembleur',
                                grade: 1,
                                logicalDelete: false,
                                cuid: 'assembly-1',
                                _id: '5a17f595977b356a3d6d391e'
                            }
                        ]
                    },
                    {
                        name: '2BIN',
                        grade: 2,
                        courses: [
                            {
                                name: 'Anglais',
                                grade: 2,
                                logicalDelete: false,
                                cuid: 'anglais-2',
                                _id: '5a17f595977b356a3d6d391d'
                            },
                            {
                                name: 'Web',
                                grade: 2,
                                logicalDelete: false,
                                cuid: 'web-2',
                                _id: '5a17f595977b356a3d6d391c'
                            },
                            {
                                name: 'Language C',
                                grade: 2,
                                logicalDelete: false,
                                cuid: 'c-lang-1',
                                _id: '5a17f595977b356a3d6d391b'
                            }
                        ]
                    },
                    {
                        name: '3BIN',
                        grade: 3,
                        courses: [
                            {
                                name: 'Anglais',
                                grade: 3,
                                logicalDelete: false,
                                cuid: 'anglais-3',
                                _id: '5a17f595977b356a3d6d391a'
                            },
                            {
                                name: 'Web',
                                grade: 3,
                                logicalDelete: false,
                                cuid: 'web-3',
                                _id: '5a17f595977b356a3d6d3919'
                            }
                        ]
                    }
                ]
            },
            {
                _id: '5a17f595977b356a3d6d3921',
                acronym: 'BIM',
                cuid: 'division-bim',
                logicalDelete: false,
                __v: 0,
                courses: [
                    {
                        name: 'Anglais',
                        grade: 1,
                        logicalDelete: false,
                        cuid: 'anglais-1',
                        _id: '5a17f595977b356a3d6d3929'
                    },
                    {
                        name: 'Web',
                        grade: 1,
                        logicalDelete: false,
                        cuid: 'web-1',
                        _id: '5a17f595977b356a3d6d3928'
                    },
                    {
                        name: 'Assembleur',
                        grade: 1,
                        logicalDelete: false,
                        cuid: 'assembly-1',
                        _id: '5a17f595977b356a3d6d3927'
                    },
                    {
                        name: 'Anglais',
                        grade: 2,
                        logicalDelete: false,
                        cuid: 'anglais-2',
                        _id: '5a17f595977b356a3d6d3926'
                    },
                    {
                        name: 'Web',
                        grade: 2,
                        logicalDelete: false,
                        cuid: 'web-2',
                        _id: '5a17f595977b356a3d6d3925'
                    },
                    {
                        name: 'Language C',
                        grade: 2,
                        logicalDelete: false,
                        cuid: 'c-lang-1',
                        _id: '5a17f595977b356a3d6d3924'
                    },
                    {
                        name: 'Anglais',
                        grade: 3,
                        logicalDelete: false,
                        cuid: 'anglais-3',
                        _id: '5a17f595977b356a3d6d3923'
                    },
                    {
                        name: 'Web',
                        grade: 3,
                        logicalDelete: false,
                        cuid: 'web-3',
                        _id: '5a17f595977b356a3d6d3922'
                    }
                ],
                grades: [
                    null,
                    {
                        name: '1BIM',
                        grade: 1,
                        courses: [
                            {
                                name: 'Anglais',
                                grade: 1,
                                logicalDelete: false,
                                cuid: 'anglais-1',
                                _id: '5a17f595977b356a3d6d3929'
                            },
                            {
                                name: 'Web',
                                grade: 1,
                                logicalDelete: false,
                                cuid: 'web-1',
                                _id: '5a17f595977b356a3d6d3928'
                            },
                            {
                                name: 'Assembleur',
                                grade: 1,
                                logicalDelete: false,
                                cuid: 'assembly-1',
                                _id: '5a17f595977b356a3d6d3927'
                            }
                        ]
                    },
                    {
                        name: '2BIM',
                        grade: 2,
                        courses: [
                            {
                                name: 'Anglais',
                                grade: 2,
                                logicalDelete: false,
                                cuid: 'anglais-2',
                                _id: '5a17f595977b356a3d6d3926'
                            },
                            {
                                name: 'Web',
                                grade: 2,
                                logicalDelete: false,
                                cuid: 'web-2',
                                _id: '5a17f595977b356a3d6d3925'
                            },
                            {
                                name: 'Language C',
                                grade: 2,
                                logicalDelete: false,
                                cuid: 'c-lang-1',
                                _id: '5a17f595977b356a3d6d3924'
                            }
                        ]
                    },
                    {
                        name: '3BIM',
                        grade: 3,
                        courses: [
                            {
                                name: 'Anglais',
                                grade: 3,
                                logicalDelete: false,
                                cuid: 'anglais-3',
                                _id: '5a17f595977b356a3d6d3923'
                            },
                            {
                                name: 'Web',
                                grade: 3,
                                logicalDelete: false,
                                cuid: 'web-3',
                                _id: '5a17f595977b356a3d6d3922'
                            }
                        ]
                    }
                ]
            }
        ],
        user: {}
    },mockFetchDivision = jest.fn(),
    mockUser = {},
    mockConnected=true;

describe('Navbar', () => {
    it('should render correctly', () => {
        const output = shallow(<Navbar connected={mockConnected} divisions={mockDivision} fetchDivision={mockFetchDivision} user={mockUser}/>);
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});
