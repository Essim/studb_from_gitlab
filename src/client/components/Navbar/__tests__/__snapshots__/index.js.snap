// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render correctly 1`] = `
<aside
  className="main-sidebar"
>
  <section
    className="sidebar"
  >
    <UserPanel
      user={Object {}}
    />
    <Connect(Search) />
    <ul
      className="sidebar-menu"
      data-widget="tree"
    >
      <Connect(Division)
        division={
          Object {
            "__v": 0,
            "_id": "5a17f595977b356a3d6d3918",
            "acronym": "BIN",
            "courses": Array [
              Object {
                "_id": "5a17f595977b356a3d6d3920",
                "cuid": "anglais-1",
                "grade": 1,
                "logicalDelete": false,
                "name": "Anglais",
              },
              Object {
                "_id": "5a17f595977b356a3d6d391f",
                "cuid": "web-1",
                "grade": 1,
                "logicalDelete": false,
                "name": "Web",
              },
              Object {
                "_id": "5a17f595977b356a3d6d391e",
                "cuid": "assembly-1",
                "grade": 1,
                "logicalDelete": false,
                "name": "Assembleur",
              },
              Object {
                "_id": "5a17f595977b356a3d6d391d",
                "cuid": "anglais-2",
                "grade": 2,
                "logicalDelete": false,
                "name": "Anglais",
              },
              Object {
                "_id": "5a17f595977b356a3d6d391c",
                "cuid": "web-2",
                "grade": 2,
                "logicalDelete": false,
                "name": "Web",
              },
              Object {
                "_id": "5a17f595977b356a3d6d391b",
                "cuid": "c-lang-1",
                "grade": 2,
                "logicalDelete": false,
                "name": "Language C",
              },
              Object {
                "_id": "5a17f595977b356a3d6d391a",
                "cuid": "anglais-3",
                "grade": 3,
                "logicalDelete": false,
                "name": "Anglais",
              },
              Object {
                "_id": "5a17f595977b356a3d6d3919",
                "cuid": "web-3",
                "grade": 3,
                "logicalDelete": false,
                "name": "Web",
              },
            ],
            "cuid": "cuiddivision1",
            "grades": Array [
              null,
              Object {
                "courses": Array [
                  Object {
                    "_id": "5a17f595977b356a3d6d3920",
                    "cuid": "anglais-1",
                    "grade": 1,
                    "logicalDelete": false,
                    "name": "Anglais",
                  },
                  Object {
                    "_id": "5a17f595977b356a3d6d391f",
                    "cuid": "web-1",
                    "grade": 1,
                    "logicalDelete": false,
                    "name": "Web",
                  },
                  Object {
                    "_id": "5a17f595977b356a3d6d391e",
                    "cuid": "assembly-1",
                    "grade": 1,
                    "logicalDelete": false,
                    "name": "Assembleur",
                  },
                ],
                "grade": 1,
                "name": "1BIN",
              },
              Object {
                "courses": Array [
                  Object {
                    "_id": "5a17f595977b356a3d6d391d",
                    "cuid": "anglais-2",
                    "grade": 2,
                    "logicalDelete": false,
                    "name": "Anglais",
                  },
                  Object {
                    "_id": "5a17f595977b356a3d6d391c",
                    "cuid": "web-2",
                    "grade": 2,
                    "logicalDelete": false,
                    "name": "Web",
                  },
                  Object {
                    "_id": "5a17f595977b356a3d6d391b",
                    "cuid": "c-lang-1",
                    "grade": 2,
                    "logicalDelete": false,
                    "name": "Language C",
                  },
                ],
                "grade": 2,
                "name": "2BIN",
              },
              Object {
                "courses": Array [
                  Object {
                    "_id": "5a17f595977b356a3d6d391a",
                    "cuid": "anglais-3",
                    "grade": 3,
                    "logicalDelete": false,
                    "name": "Anglais",
                  },
                  Object {
                    "_id": "5a17f595977b356a3d6d3919",
                    "cuid": "web-3",
                    "grade": 3,
                    "logicalDelete": false,
                    "name": "Web",
                  },
                ],
                "grade": 3,
                "name": "3BIN",
              },
            ],
            "logicalDelete": false,
          }
        }
        key="navbarBIN"
      />
      <Connect(Division)
        division={
          Object {
            "__v": 0,
            "_id": "5a17f595977b356a3d6d3921",
            "acronym": "BIM",
            "courses": Array [
              Object {
                "_id": "5a17f595977b356a3d6d3929",
                "cuid": "anglais-1",
                "grade": 1,
                "logicalDelete": false,
                "name": "Anglais",
              },
              Object {
                "_id": "5a17f595977b356a3d6d3928",
                "cuid": "web-1",
                "grade": 1,
                "logicalDelete": false,
                "name": "Web",
              },
              Object {
                "_id": "5a17f595977b356a3d6d3927",
                "cuid": "assembly-1",
                "grade": 1,
                "logicalDelete": false,
                "name": "Assembleur",
              },
              Object {
                "_id": "5a17f595977b356a3d6d3926",
                "cuid": "anglais-2",
                "grade": 2,
                "logicalDelete": false,
                "name": "Anglais",
              },
              Object {
                "_id": "5a17f595977b356a3d6d3925",
                "cuid": "web-2",
                "grade": 2,
                "logicalDelete": false,
                "name": "Web",
              },
              Object {
                "_id": "5a17f595977b356a3d6d3924",
                "cuid": "c-lang-1",
                "grade": 2,
                "logicalDelete": false,
                "name": "Language C",
              },
              Object {
                "_id": "5a17f595977b356a3d6d3923",
                "cuid": "anglais-3",
                "grade": 3,
                "logicalDelete": false,
                "name": "Anglais",
              },
              Object {
                "_id": "5a17f595977b356a3d6d3922",
                "cuid": "web-3",
                "grade": 3,
                "logicalDelete": false,
                "name": "Web",
              },
            ],
            "cuid": "division-bim",
            "grades": Array [
              null,
              Object {
                "courses": Array [
                  Object {
                    "_id": "5a17f595977b356a3d6d3929",
                    "cuid": "anglais-1",
                    "grade": 1,
                    "logicalDelete": false,
                    "name": "Anglais",
                  },
                  Object {
                    "_id": "5a17f595977b356a3d6d3928",
                    "cuid": "web-1",
                    "grade": 1,
                    "logicalDelete": false,
                    "name": "Web",
                  },
                  Object {
                    "_id": "5a17f595977b356a3d6d3927",
                    "cuid": "assembly-1",
                    "grade": 1,
                    "logicalDelete": false,
                    "name": "Assembleur",
                  },
                ],
                "grade": 1,
                "name": "1BIM",
              },
              Object {
                "courses": Array [
                  Object {
                    "_id": "5a17f595977b356a3d6d3926",
                    "cuid": "anglais-2",
                    "grade": 2,
                    "logicalDelete": false,
                    "name": "Anglais",
                  },
                  Object {
                    "_id": "5a17f595977b356a3d6d3925",
                    "cuid": "web-2",
                    "grade": 2,
                    "logicalDelete": false,
                    "name": "Web",
                  },
                  Object {
                    "_id": "5a17f595977b356a3d6d3924",
                    "cuid": "c-lang-1",
                    "grade": 2,
                    "logicalDelete": false,
                    "name": "Language C",
                  },
                ],
                "grade": 2,
                "name": "2BIM",
              },
              Object {
                "courses": Array [
                  Object {
                    "_id": "5a17f595977b356a3d6d3923",
                    "cuid": "anglais-3",
                    "grade": 3,
                    "logicalDelete": false,
                    "name": "Anglais",
                  },
                  Object {
                    "_id": "5a17f595977b356a3d6d3922",
                    "cuid": "web-3",
                    "grade": 3,
                    "logicalDelete": false,
                    "name": "Web",
                  },
                ],
                "grade": 3,
                "name": "3BIM",
              },
            ],
            "logicalDelete": false,
          }
        }
        key="navbarBIM"
      />
    </ul>
  </section>
</aside>
`;
